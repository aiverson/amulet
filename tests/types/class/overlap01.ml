class foo 'a begin
  val x : 'a -> ()
end
instance foo 'a begin let x _ = () end
instance foo () begin let x _ = () end
