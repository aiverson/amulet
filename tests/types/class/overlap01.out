overlap01.ml[4:1 ..4:38]: error (E2021)
  Overlapping instances for foo unit
  • Note: first defined here,
  │ 
4 │ instance foo 'a begin let x _ = () end
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   but also defined here
  │ 
5 │ instance foo () begin let x _ = () end
  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
overlap01.ml[4:1 ..4:38]: Use amc explain 2021 for more information
